File: clio-j/core/src/test/java/clio/core/TestNumbers.java
package clio.core;

import clio.core.Numbers;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class TestNumbers {
    @Test
    public void testDeviation() {
        assertEquals(1.0, Numbers.deviation(0, 10), 0.001);
        assertEquals(1.0, Numbers.deviation(10, 0), 0.001);
        assertEquals(0.0, Numbers.deviation(10, 10), 0.001);
        assertEquals(0.4, Numbers.deviation(10, 6), 0.001);
        assertEquals(0.4, Numbers.deviation(6, 10), 0.001);
        assertEquals(1.4, Numbers.deviation(-10, 4), 0.001);
        assertEquals(1.4, Numbers.deviation(-4, 10), 0.001);
    }

    @Test
    public void testFloorToZero() {
        assertEquals(10, Numbers.floorToZero(10.1));
        assertEquals(12, Numbers.floorToZero(12.9));
        assertEquals(-10, Numbers.floorToZero(-10.1));
        assertEquals(-12, Numbers.floorToZero(-12.9));

        assertEquals(11100, Numbers.floorToZero(11111, 100));
        assertEquals(11200, Numbers.floorToZero(11299, 100));
        assertEquals(-11100, Numbers.floorToZero(-11111, 100));
        assertEquals(-11200, Numbers.floorToZero(-11299, 100));
    }
}

File: clio-py/tests/clio/test_numbers.py
from clio.numbers import floor_to_zero, ceil_to_zero


def test_floor_to_zero():
    assert 10 == floor_to_zero(10.1)
    assert 12 == floor_to_zero(12.9)
    assert -10 == floor_to_zero(-10.1)
    assert -12 == floor_to_zero(-12.9)

    assert 11100 == floor_to_zero(11111, step=100)
    assert 11200 == floor_to_zero(11299, step=100)
    assert -11100 == floor_to_zero(-11111, step=100)
    assert -11200 == floor_to_zero(-11299, step=100)

    assert 11 == floor_to_zero(10.999, epsilon=0.01)
    assert -11 == floor_to_zero(-10.999, epsilon=0.01)


def test_ceil_to_zero():
    assert 11 == ceil_to_zero(10.1)
    assert 13 == ceil_to_zero(12.9)
    assert -11 == ceil_to_zero(-10.1)
    assert -13 == ceil_to_zero(-12.9)

    assert 11200 == ceil_to_zero(11111, step=100)
    assert 11300 == ceil_to_zero(11299, step=100)
    assert -11200 == ceil_to_zero(-11111, step=100)
    assert -11300 == ceil_to_zero(-11299, step=100)

    assert 11 == ceil_to_zero(11.0001, epsilon=0.01)
    assert -11 == ceil_to_zero(-11.0001, epsilon=0.01)
